### PR TITLE
Add auth errors, auth error tests

### PR DESCRIPTION
**Pull request by tschuy**
Super simple.

### CODE REVIEW

File: tests/errors.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        });
+    });
+
+    describe('7: errorAuthenticationFailure', function() {
+        it('returns authentication failure block', function(done) {
+            err = errors.errorAuthenticationFailure('Invalid key');
+            expect(err.status).to.equal(401);
+            expect(err.error).to.equal('Authentication failure');
+            expect(err.text).to.equal('Invalid key');
+            done();
```
Comment (iankronquist) [inline tests/errors.js]: This seems excessive to me. If we look at the logic which is being tested, we're really just testing the createError function.

**Comment by ElijahCaine** [comment]
@tschuy Tests pass, everything seems pretty solid here.
As far as @iankronquist's comment, I think adding designated createError tests would be beneficial, however as it stands this gets the job done. createError tests ought to be a separate pull request so I'm gonna give this a +1.

**Comment by ElijahCaine** [comment]
@tschuy Tests pass, everything seems pretty solid here.
As far as @iankronquist's comment, I think adding designated createError tests would be beneficial, however as it stands this gets the job done. createError tests ought to be a separate pull request so I'm gonna give this a +1.

**Comment by iankronquist** [comment]
Looks good +1.

**Pull request merged**